esiclient/utils.py

```diff
@@ -25,33 +25,39 @@ def get_network_display_name(network):
         return network.name
 
 
-def get_network_info_from_port(port, client):
+def get_network_info_from_port(port, client, networks_dict=None):
     """Return Neutron network name and ips from port
 
     :param port: a Neutron port
+    :param networks_dict: networks dict {id:network}
     :param client: neutron client
     """
-    network = client.get_network(port.network_id)
+    if networks_dict:
+        network = networks_dict.get(port.network_id)
+    else:
+        network = client.get_network(port.network_id)
     fixed_ip = ''
     if port.fixed_ips and len(port.fixed_ips) > 0:
         fixed_ip = port.fixed_ips[0]['ip_address']
 
     return get_network_display_name(network), fixed_ip
 
 
-def get_full_network_info_from_port(port, client):
+def get_full_network_info_from_port(port, client, networks_dict=None):
```

Comment:
If you take the suggestion above then networks_dict here would also default to {}